addons/im_livechat/static/tests/channel_join_leave.test.js

```diff
@@ -0,0 +1,51 @@
+const test = QUnit.test; // QUnit.test()
+
+import { serverState, startServer } from "@bus/../tests/legacy/helpers/mock_python_environment";
+import { TEST_GROUP_IDS } from "@bus/../tests/legacy/helpers/test_constants";
+import { openDiscuss, start } from "@mail/../tests/legacy/helpers/test_utils";
+import { assertSteps, click, contains, step } from "@web/../tests/legacy/utils";
+import { patchWithCleanup, triggerHotkey } from "@web/../tests/legacy/helpers/utils";
+import { Command } from "@mail/../tests/helpers/command";
+
+QUnit.module("channel join/leave");
+
+test("from the discuss app", async () => {
+    const pyEnv = await startServer();
+    pyEnv["res.users"].write([serverState.userId], {
+        groups_id: Command.link(TEST_GROUP_IDS.groupLivechatId),
+    });
+    pyEnv["im_livechat.channel"].create({ name: "HR", user_ids: [serverState.userId] });
+    const { env } = await start();
+    patchWithCleanup(env.services.notification, { add: (message) => step(message) });
```

Comment:
for notification to need for a patch, you can check the actual notification in the DOM